src/Tribe/Tickets_View.php

```diff
@@ -57,18 +57,49 @@ public static function hook() {
 		return $myself;
 	}
 
+	/**
+	 * Tries to Flush the Rewrite rules
+	 *
+	 * @return void
+	 */
+	public function maybe_regenerate_rewrite_rules() {
+		// if they don't have any rewrite rules, do nothing
+		// Don't try to run stuff for non-logged users (too time consuming)
+		if ( ! is_array( $GLOBALS['wp_rewrite']->rules ) || ! is_user_logged_in() ) {
```

Comment:
Normally `$wp_rewrite->rules` is going to be in memory already so this doesn't represent a large overhead and I'm not super-concerned, but might it be preferable to do this on activation (and on plugin update) rather than every request? Just a thought, don't necessarily need to change it.